CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+# Contributing
+
+This document provides guidelines for contributing to samples.
+
+## Dependencies
+
+The following dependencies must be installed on the development system:
+
+- [Docker Engine][docker-engine]
+- [Google Cloud SDK][google-cloud-sdk]
+- [make]
+
+## Linting and Formatting
+Files in the repository are linted or formatted to maintain a standard of quality and statically validated.
+You can run this check locally:
+
+```
+cd terraform-docs-samples
+make docker_test_lint
+```
+
+## Integration Testing
+
+Integration tests are used to verify that the samples are actuatable
+by Terraform. Tests are dynamically discovered based on directories in
+repo root and executed using the [blueprint-test](https://pkg.go.dev/github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test) framework.
+
+### Test Environment
+
+The easiest way to test the samples are in isolated test projects. The
+setup for projects are defined in [test/setup](./test/setup/)
+directory.
+
+To use this setup, you need to authenticate gcloud with an identity that has
+Project Creator access on a folder and the Billing Account User role on a billing account.
+
+Set the following environment variables.
+```
+export TF_VAR_org_id="your_org_id"
+export TF_VAR_folder_id="your_folder_id"
+export TF_VAR_billing_account="your_billing_account_id"
+```
+
+Ensure that you are locally authenticated using gcloud:
+
+```
+gcloud auth application-default login
+```
+
+With these settings in place, you can prepare a test project using Docker:
+
+```
+cd terraform-docs-samples
+make docker_test_prepare
```

Comment:
Can we add sample output here?

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

project_ids = [
  "ci-tf-samples-0-c5ab",
  "ci-tf-samples-1-f344",
  "ci-tf-samples-2-b5a7",
  "ci-tf-samples-3-6eda",
]
sa_key = <sensitive>
```

Reply:
Added sample output, user need not know to know which project since the test framework handles the selection.